tsdb/engine/tsm1/compact.go

```diff
@@ -1208,7 +1208,7 @@ func (k *tsmKeyIterator) combine(dedup bool) blocks {
 }
 
 func (k *tsmKeyIterator) chunk(dst blocks) blocks {
-	for len(k.mergedValues) > k.size {
+	if len(k.mergedValues) > k.size {
```

Comment:
This should be an `if` as you have it.  When the return was added, I forgot to switch it from a loop.